app/src/test/java/com/example/trainingintive/dog_images_feature/presentation/DogImageViewModelTest.kt

```diff
@@ -36,20 +40,27 @@ class DogImageViewModelTest : RulesForTests() {
 
     @Test
     fun `after successfully download, dogimage should be inserted into local database`() {
-        DogImageViewModel(repository, schedulersForTests)
+        DogImageViewModel(repository, schedulersForTests, mainNavigator)
 
         verify { repository.insert(sampleDog) }
     }
 
     @Test
-    fun `if repository returns error livedata should contain message`() {
-        val errorMessage = "Unable to resolve host \"dog.ceo\": No address associated with hostname"
-        every { repository.getDogImageUrl() } returns Single.error(Exception(errorMessage))
+    fun `if repository returns error vm should send error event`() {
+        every { repository.getDogImageUrl() } returns Single.error(exception)
 
-        val tested = DogImageViewModel(repository, schedulersForTests)
+        DogImageViewModel(repository, schedulersForTests, mainNavigator)
 
-        tested.dog.observeForever { result ->
-            result shouldBeEqualTo errorMessage
-        }
+        verify { mainNavigator.sendEvent(ofType(MainScreenEvent.Error::class)) }
+    }
+
+    @Test
+    fun `if repository returns error vm should call toErrorTextId method`() {
+        mockkStatic(exception::toErrorTextId)
```

Comment:
You also need to unmock it because it could affect other tests. You can mockk in @Before function and unmock in @After function